airbyte-integrations/connectors/source-freshdesk/source_freshdesk/streams.py

```diff
@@ -0,0 +1,254 @@
+        # with default value 30 days look back.
+        self.start_date = pendulum.parse(start_date) if start_date else pendulum.now() - pendulum.duration(days=30)
+    
+    @property
+    def url_base(self) -> str:
+        return parse.urljoin(f"https://{self.domain.rstrip('/')}", "/api/v2")
+    
+    def backoff_time(self, response: requests.Response) -> Optional[float]:
+        if response.status_code == requests.codes.too_many_requests:
+            return float(response.headers.get("Retry-After"))
+    
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        try:
+            link_header = response.headers.get("Link")
+            if not link_header:
+                return {}
+            match = self.link_regex.search(link_header)
+            next_url = match.group(1)
+            params = parse.parse_qs(parse.urlparse(next_url).query)
+            return self.parse_link_params(link_query_params=params)
+        except Exception as e:
+            raise KeyError(f"error parsing next_page token: {e}")
+    
+    def parse_link_params(self, link_query_params: Mapping[str, List[str]]) -> Mapping[str, Any]:
+        return {"per_page": link_query_params['per_page'][0], "page": link_query_params['page'][0]}
+    
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        params = {"per_page": self.result_return_limit}
+        if next_page_token and "page" in next_page_token:
+            params["page"] = next_page_token["page"]
+        return params
+    
+    def _consume_credit(self, credit):
+        """Consume call credit, if there is no credit left within current window will sleep til next period"""
+        if self._call_credit:
+            self._call_credit.consume(credit)
+    
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping[str, Any]]:
+        self._consume_credit(self.call_credit)
+        yield from super().read_records(sync_mode=sync_mode, cursor_field=cursor_field, stream_slice=stream_slice, stream_state=stream_state)
+    
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        data = response.json()
+        if not data:
+            return []
+        for element in data:
+            yield element
+
+
+class IncrementalFreshdeskStream(FreshdeskStream, IncrementalMixin):
+
+    state_filter = "updated_since"  # Name of filter that corresponds to the state
+
+    def __init__(self, authenticator: AuthBase, config: Mapping[str, Any], *args, **kwargs):
+        super().__init__(authenticator=authenticator, config=config, *args, **kwargs)
+        self._state = None
```

Comment:
I don't think you should use `_.state` class attribute if you expose the `state` property. You should rather manage a `self._cursor_value` attribute

Reply:
Makes sense. Updated.